pinot-core/src/main/java/org/apache/pinot/core/operator/filter/FilterOperatorUtils.java

```diff
@@ -28,9 +28,207 @@
 
 
 public class FilterOperatorUtils {
+
+  private static volatile Implementation _instance = new Implementation();
```

Comment:
`volatile` might introduce extra overhead since the util will be accessed by multiple threads. We will always plug-in the implementation at service start time, so it should be okay to remove the `volatile`

Reply:
It doesn't. Volatile is expensive in x86 if and only if there are writes and reads. In a scenario like this one, where the attribute will be changed once at startup. Also, this code is not in the hotpath (row execution) but in the planning phase. There are quite more expensive costs like our heavily usage of allocation in the planning phase.

Anyway, we can remove it if you prefer that.

References: 
* https://stackoverflow.com/questions/4633866/is-volatile-expensive

Reply:
This is good learning for me. I thought reading volatile is always from main memory, but seems it can be read from L1 cache as long as the value is not written to for most modern hardware